### PR TITLE
⚠️ Use consistent naming for API constants

### PR DESCRIPTION
**Pull request by chiukapoor**
Signed-off-by: Chirayu Kapoor <dev.csociety@gmail.com>

**What this PR does / why we need it**:
Fixes the inconsistent constant names

**Which issue(s) this PR fixes** 
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/7611

### CODE REVIEW

File: api/v1alpha3/common_types.go [hunk 1]
```diff
@@ -22,9 +22,9 @@ import (
 )
 
 const (
-	// ClusterLabelName is the label set on machines linked to a cluster and
+	// ClusterNameLabel is the label set on machines linked to a cluster and
```
Comment (sbueringer) [inline api/v1alpha3/common_types.go:25]: Please only rename the constants in the latest apiVersion (v1beta1, not v1alpha3/v1alpha4)

File: api/v1beta1/common_types.go [hunk 2]
```diff
@@ -22,9 +22,9 @@ import (
 )
 
 const (
-	// ClusterLabelName is the label set on machines linked to a cluster and
+	// ClusterNameLabel is the label set on machines linked to a cluster and
 	// external objects(bootstrap and infrastructure providers).
-	ClusterLabelName = "cluster.x-k8s.io/cluster-name"
+	ClusterNameLabel = "cluster.x-k8s.io/cluster-name"
```
Comment (sbueringer) [inline api/v1beta1/common_types.go:27]: The renames should be eventually all added to the migration doc.

The migration doc should be created by the release team in 1-2 weeks: https://github.com/kubernetes-sigs/cluster-api/blob/d146f6d9e264241c5db29e8ecf99a390ab97807d/docs/release/release-tasks.md#add-docs-to-collect-release-notes-for-users-and-migration-notes-for-provider-implementers

Theoratically we could do it here already but I would like to give the first release team a chance to try out our new release process, ...

Comment (sbueringer) [reply api/v1beta1/common_types.go:27]: Let's also in general hold merging this PR until at least after the v1.3.0 release (1st December) just to ensure it doesn't interfere with the release / cherry-picks until then.

**Comment by sbueringer** [comment]
In general looks fine +/- the comments above.

Given the amount of files that have to be touched for those renames. Maybe you can do further renames for now in separate commits and then we can see how big the impact overall is and potentially decide to move individual commits to separate PRs?

Although this label should be pretty much the one with the most usage.

File: api/v1alpha3/common_types.go (hunk 1 above)
Comment (chiukapoor) [reply api/v1alpha3/common_types.go:25]: Understood, will update the code to rename only for v1beta1.

Q) What about the usage of constant `ClusterLabelName` in files inside `bootstrap/kubeadm/internal`, `cmd/clusterctl`, `controller/external`, `controlplane/kubeadm/internal` etc. Basically every other file apart from `api/*`. I have changed it for now, but is it required to do so?

File: api/v1beta1/common_types.go (hunk 2 above)
Comment (chiukapoor) [reply api/v1beta1/common_types.go:27]: Sure, will make sure the PR is on hold until at least after the v1.3.0 release.

File: api/v1alpha3/common_types.go (hunk 1 above)
Comment (sbueringer) [reply api/v1alpha3/common_types.go:25]: Yes it is and that's fine

**Comment by sbueringer** [comment]
Sounds perfect, thank you!

File: api/v1alpha3/common_types.go (hunk 1 above)
Comment (chiukapoor) [reply api/v1alpha3/common_types.go]: Regarding the usage of v1alpha3/v1alpha4, We are using some types and structs from `v1alpha3/v1alpha4` but no constants.

**Comment by chiukapoor** [comment]
Updated the constant from `ClusterTopologyMachineDeploymentLabelName' to `ClusterTopologyMachineDeploymentNameLabel`

**Comment by chiukapoor** [comment]
Updated the constant from `ProviderLabelName` to `ProviderLabel`, executed the test using `make test` [passed]

**Comment by chiukapoor** [comment]
While working on label `ProvideLabel`, the file `hack/tools/tilt-prepare/main.go` was updated. Do we need to run `make tilt-prepare` to update the Tiltfile?

File: api/v1alpha3/common_types.go (hunk 1 above)
Comment (sbueringer) [reply api/v1alpha3/common_types.go]: Took a look those usages of v1alpha3/v1alpha4 are fine

Comment (sbueringer) [reply api/v1alpha3/common_types.go]: Changes lgtm to so far

**Comment by sbueringer** [comment]
> While working on label `ProvideLabel`, the file `hack/tools/tilt-prepare/main.go` was updated. Do we need to run `make tilt-prepare` to update the Tiltfile?

No that's fine. The tiltfile is using the tilt-prepare binary at runtime. tilt-prepare does not generate the TIltfile.

File: api/v1beta1/common_types.go [hunk 3]
```diff
@@ -45,18 +45,18 @@ const (
...
 	// ClusterTopologyUnsafeUpdateClassNameAnnotation can be used to disable the webhook check on
 	// update that disallows a pre-existing Cluster to be populated with Topology information and Class.
 	ClusterTopologyUnsafeUpdateClassNameAnnotation = "unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check"
 
-	// ProviderLabelName is the label set on components in the provider manifest.
+	// ProviderLabel is the label set on components in the provider manifest.
 	// This label allows to easily identify all the components belonging to a provider; the clusterctl
 	// tool uses this label for implementing provider's lifecycle operations.
-	ProviderLabelName = "cluster.x-k8s.io/provider"
+	ProviderLabel = "cluster.x-k8s.io/provider"
```
Comment (fabriziopandini) [inline api/v1beta1/common_types.go:59]: q: should this be 
```suggestion
	ProviderNameLabel = "cluster.x-k8s.io/provider"
```
@sbueringer opinions?

**Review by fabriziopandini — COMMENTED** [review]
overall lgtm from my side

File: api/v1beta1/common_types.go (hunk 3 above)
Comment (sbueringer) [reply api/v1beta1/common_types.go:59]: I would go with
>I would suggest to only include a Name in the constant name if the label actually contains name. In that case we should include it before Label and not after.
> https://github.com/kubernetes-sigs/cluster-api/issues/7611

If the label itself is wrong and should be `cluster.x-k8s.io/provider-name` that is something that should be changed independently.

I would keep the mapping from label string to constant name simple and if our label strings are wrong let's fix them if we think it's worth the effort.

**Comment by chiukapoor** [comment]
Hello @sbueringer and @fabriziopandini, I have fixed all the constants with inconsistency mentioned in https://github.com/kubernetes-sigs/cluster-api/issues/7611 and have made the requested change of `ProviderLabel` to `ProviderNameLabel`.

PTAL

**Comment by fabriziopandini** [comment]
Please squash commits, I will take a look right after

**Comment by chiukapoor** [comment]
The commits have been squashed, I have also fixed the below mismatch in the comment and function name.

https://github.com/kubernetes-sigs/cluster-api/blob/709d848d6a798553c11b6a22229c5ec3e2a0dd9f/internal/controllers/topology/cluster/cluster_controller_test.go#L1040-L1042

Also, as per the update from @sbueringer in reply https://github.com/kubernetes-sigs/cluster-api/pull/7618#discussion_r1031998177, the PR merge needs to be on hold until 1.3.0 release.

File: api/v1beta1/common_types.go [hunk 4]
```diff
@@ -45,18 +45,18 @@ const (
...
 	// ClusterTopologyUnsafeUpdateClassNameAnnotation can be used to disable the webhook check on
 	// update that disallows a pre-existing Cluster to be populated with Topology information and Class.
 	ClusterTopologyUnsafeUpdateClassNameAnnotation = "unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check"
 
-	// ProviderLabelName is the label set on components in the provider manifest.
+	// ProviderNameLabel is the label set on components in the provider manifest.
 	// This label allows to easily identify all the components belonging to a provider; the clusterctl
 	// tool uses this label for implementing provider's lifecycle operations.
-	ProviderLabelName = "cluster.x-k8s.io/provider"
+	ProviderNameLabel = "cluster.x-k8s.io/provider"
```
Comment (fabriziopandini) [inline api/v1beta1/common_types.go:59]: As per @sbueringer this should be 
```suggestion
	ProviderLabel = "cluster.x-k8s.io/provider"
```
(the label is just provider, not provider-name)

**Review by fabriziopandini — COMMENTED** [review]
One last nit from my side
Also, @sbueringer what about marking this PR as ⚠️ so it will pop up in the 1.4 release notes

File: api/v1beta1/common_types.go (hunk 4 above)
Comment (chiukapoor) [reply api/v1beta1/common_types.go:59]: As per the @sbueringer suggestion in reply https://github.com/kubernetes-sigs/cluster-api/pull/7618#discussion_r1034255049 I made this exception. Please do let me know if I should revert it?

Comment (sbueringer) [reply api/v1beta1/common_types.go:59]: I'm fine with `ProviderNameLabel` for the stated reason. @fabriziopandini I'm also fine with `ProviderLabel` if you prefer it.

**Comment by sbueringer** [comment]
> One last nit from my side Also, @sbueringer what about marking this PR as ⚠️ so it will pop up in the 1.4 release notes

Yup let's do that.

/retitle ⚠️ Use consistent naming for API constants

**Comment by sbueringer** [comment]
/lgtm

I think we can merge this before the migration doc. Let's just keep the corresponding issue open and do a follow-up PR for: https://github.com/kubernetes-sigs/cluster-api/pull/7618?notification_referrer_id=NT_kwDOAEckWLI0OTIxNDA3OTc1OjQ2NjIzNjA&notifications_query=repo%3Akubernetes-sigs%2Fcluster-api#discussion_r1031997700

**Comment by chiukapoor** [comment]
> Let's just keep the corresponding issue open and do a follow-up PR for: https://github.com/kubernetes-sigs/cluster-api/pull/7618?notification_referrer_id=NT_kwDOAEckWLI0OTIxNDA3OTc1OjQ2NjIzNjA&notifications_query=repo%3Akubernetes-sigs%2Fcluster-api#discussion_r1031997700

@sbueringer should I create another PR attached to https://github.com/kubernetes-sigs/cluster-api/issues/7611 for the migration doc? If yes, what should be added to the doc?

Also, how may I resolve the `pull-cluster-api-apidiff-main` test?

**Comment by sbueringer** [comment]
> @sbueringer should I create another PR attached to https://github.com/kubernetes-sigs/cluster-api/issues/7611 for the migration doc? If yes, what should be added to the doc?

Let's just add a comment to #7611 for now and make sure that the current PR does not closes the issue (change "fixes" in the pr description to "Part of")

> Also, how may I resolve the pull-cluster-api-apidiff-main test?

You don't have to resolve it. It's just an indiciation for reviewers that we change an API. It's optional for merge.

**Pull request merged**